cmake/modules/CheckCompiler.cmake

```diff
@@ -201,6 +201,19 @@ endif()
 
 
 #---Setup details depending on the major platform type----------------------------------------------
+
+function(ROOT_SET_COMPILER_FLAGS VARNAME FLAGS DOCSTRING)
+    # Don't set ROOT's defaults if it's overwritten, e.g. via setting CMAKE_CXX_FLAGS_DEBUG
+    # The *_INIT versions may have additional whitespaces.
+    STRING(STRIP ${${VARNAME}} COMPARE_VAR)
+    STRING(STRIP ${${VARNAME}_INIT} COMPARE_VAR_INIT)
+    if (COMPARE_VAR STREQUAL COMPARE_VAR_INIT)
+        set(${VARNAME} "${FLAGS}" CACHE STRING "${DOCSTRING}" FORCE)
```

Comment:
What happens if you reconfigure? Probably what we want (= it doesn't get overwritten again), but did you check?

Reply:
Here's what happens, it overwrites on reconfigure:

```
# Initial cmake call
/usr/bin/cmake3 -DCMAKE_CXX_FLAGS_RELEASE="foo" /path/to/src
-- Build Type: 'Release' (flags = 'foo')
# Reconfigure
/usr/bin/cmake3 -DCMAKE_CXX_FLAGS_RELEASE="bar" /path/to/src
-- Build Type: 'Release' (flags = 'bar')
# Reconfigure
/usr/bin/cmake3 /path/to/src
-- Build Type: 'Release' (flags = '-O3')
```

I don't see how we can detect a reconfigure in this setup.